{Doc} Refine extension release notes (#13960)
Azure/azure-cli: doc/extensions/authoring.md

### Publish

> **For the extension that source code is hosted in** [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb), we will help to release once your code is merged into master branch. You must not update [index.json](https://github.com/Azure/azure-cli-extensions/blob/master/src/index.json) manually in this case.

Reviewer: ```suggestion
**For the extension whose source code is hosted in [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb)**, we will release for you once your code is merged into `master` branch. You must not update [index.json](https://github.com/Azure/azure-cli-extensions/blob/master/src/index.json) manually in this case.
```

> We detect Python package version via `python setup.py --version`. Only when the version is upgraded, the release process is triggered to help you build and upload the extension WHL file, then update the `index.json` automatically. Subsequently, a PR with newer extension info will be created to update `index.json`, we will merge it once CI is passed. Then, a newer extension is published.

Reviewer: ```suggestion
We detect Python package version via `python setup.py --version`. Only when the version is upgraded, the release process is triggered to help you build and upload the extension WHL file, then update the `index.json` automatically. Subsequently, a PR with newer extension info will be created to update `index.json`, we will merge it once CI passes. Then, the new extension is published.
```

> **For the extension that source code is not hosted in** [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb), you need to build and upload the WHL file to a public location and optionally advertise the new extension in the repo's index.json file for discoverability. For public extensions that are published to a storage account, the following command will accomplish all of this.

Reviewer: ```suggestion
**For the extension that source code is not hosted in** [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb), you need to build and upload the WHL file to a public location and optionally advertise the new extension in the repo's `README.md` file for discoverability. For public extensions that are published to a storage account, the following command will accomplish all of this.
```

Author: It's `index.json` to let new extension be discoverable.

> **For the extension that source code is not hosted in** [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb), you need to build and upload the WHL file to a public location and optionally advertise the new extension in the repo's index.json file for discoverability. For public extensions that are published to a storage account, the following command will accomplish all of this.

Reviewer: ```suggestion
**For the extension that source code is not hosted in [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb)**, you need to build and upload the WHL file to a public location and optionally advertise the new extension in the repo's index.json file for discoverability. For public extensions that are published to a storage account, the following command will accomplish all of this.
```

`azdev extension publish <NAME> --update-index [--storage-account NAME --storage-container NAME --storage-subscription GUID]`

The storage fields can be stored in your config file or as environment variables so you need not supply them every time. Once the publish command has been run (you must be logged in to the Azure CLI for it to succeed), you can open a PR that will contain your code changes and the index update. This used to be done in two steps.

Once your extension is published, you can view it via `az extension list-avaliable -o table`.

> However, if you want your extension listed in [Official Available Extensions for Azure CLI](https://docs.microsoft.com/en-us/cli/azure/azure-cli-extensions-list?view=azure-cli-latest), you have to wait until next [Azure CLI release](https://github.com/Azure/azure-cli/milestones). We will update that documantion everytime Azure CLI is released. Or, you could fire an PR to update it manually if it's urgent.

Reviewer: ```suggestion
However, if you want your extension to be listed in [Official Available Extensions for Azure CLI](https://docs.microsoft.com/en-us/cli/azure/azure-cli-extensions-list?view=azure-cli-latest), you have to wait until the next [Azure CLI release](https://github.com/Azure/azure-cli/milestones). We update that document every time Azure CLI is released. Alternatively, you could file a PR to update it manually if it's urgent.
```

Also, see the [FAQ](faq.md).

> ### What's differencies between host and not host source code in Azure/azure-cli-extensions

Reviewer: ```suggestion
### What's differencies between hosting and not hosting source code in Azure/azure-cli-extensions
```

> ### What's differencies between hosting and not hosting source code in Azure/azure-cli-extensions

Reviewer: ```suggestion
### Differences between hosting and not hosting source code in Azure/azure-cli-extensions
```

> An advantage that hosting extension in [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb) is that you could save the time to repeatedly build and upload the WHL file to pass some CI checks.

Reviewer: ```suggestion
An advantage of hosting extension in [Azure/azure-cli-extensions](https://github.com/Azure/azure-cli-extensionsb) is that you could save the time to repeatedly build and upload the WHL file to pass CI checks.
```

> As for hosting outside, you could easily provide a direct and explicit entry to end user with your customized introduction page, like [Azure DevOps Extension for Azure CLI](https://github.com/Azure/azure-devops-cli-extension), which is fit to popularize your Azure CLI extension if you plan to.

Reviewer: ```suggestion
As for hosting outside, you could easily provide a direct and explicit entry to end users with your customized introduction page, like [Azure DevOps Extension for Azure CLI](https://github.com/Azure/azure-devops-cli-extension), which is fit to popularize your Azure CLI extension if you plan to.
```

> Otherwise, user have to go deeper to find the detail page in Azure/azure-cli-extensions, such as `src/azure-firewall`, `src/vm-repair`.

Reviewer: ```suggestion
Otherwise, users have to go deeper to find the detailed page in Azure/azure-cli-extensions, such as `src/azure-firewall`, `src/vm-repair`.
```
